### ISSUE TITLE
MA0011 is not reported for parameterless ToString unless ToString(IFormatProvider formatProvider) overload exists

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#895]
Right now, **MA0011** is not reported for use of a parameterless `ToString()` on a custom type unless that type also has a `ToString(IFormatProvider formatProvider)` overload:

```c#
using System;

internal sealed class Program
{
    private static void Main(string[] args)
    {
        Foo foo = new Foo();

        Console.WriteLine(foo.ToString()); // MA0011 is not reported unless you uncomment the ToString(IFormatProvider formatProvider) overload
    }
}

public sealed class Foo : IFormattable
{
    public string ToString(string format)
    {
        return string.Empty;
    }

    /*
    public string ToString(IFormatProvider formatProvider)
    {
        return string.Empty;
    }
    */

    public string ToString(string format, IFormatProvider formatProvider)
    {
        return string.Empty;
    }

    public override string ToString()
    {
        return string.Empty;
    }
}
```

Wouldn't it better to also report **MA0011** if the type implements **IFormattable** (or defines a `ToString(string format, IFormatProvider formatProvider)` overload?

**Comment by meziantou** [comment]
Should be fixed by e17b7630450225711d9a618de76ca9b960792615
